src/garage/envs/environment.py

```diff
@@ -0,0 +1,176 @@
+                    The initial observation.
+                * terminal (bool): The termination signal for the this time
+                    step.
+                * env_info (dict): A dict arbitrary environment state
+                    information.
+                * agent_info (numpy.ndarray): A dict of arbitrary agent
+                    state information.
+                * step_type (garage.StepType): a `StepType` enum of value
+                    StepType.FIRST.
+
+        """
+
+    @abc.abstractmethod
+    def step(self, action):
+        """Steps the environment using to the action and returns a `TimeStep`.
+
+        If the environment returned a `TimeStep` with `StepType.LAST` at the
+        previous step, this call to `step` will start a new sequence and
+        `action` will be ignored.
+
+        This method will also start a new sequence if called after the
+        environment has been constructed and `reset` has not been called.
+        Again, in this case `action` will be ignored.
+
+        Args:
+            action (np.ndarray): A NumPy array, or a nested dict, list or tuple
+                of arrays corresponding to `action_spec()`.
+
+        Returns:
+            garage.TimeStep: The TimeStep after the action. Attributes:
+                * env_spec (garage.envs.EnvSpec): Environment specification.
+                * observation: an `np.ndarray`, or a nested dict, list or
+                    tuple corresponding to :obj:`env_spec.observation_space`.
+                    The observation before applying the action.
+                * action (np.ndarray): A NumPy array, or a nested dict, list or
+                    tuple of arrays corresponding to `action_spec()`.
+                * reward (float): The reward for taking the action given the
+                    observation, at the this time step.
+                * next_observation: an `np.ndarray`, or a nested dict, list or
+                    tuple corresponding to :obj:`env_spec.observation_space`.
+                    The observation after applying the action.
+                * terminal (bool): The termination signal for the this time
+                    step.
+                * env_info (dict): A dict arbitrary environment state
+                    information.
+                * agent_info (numpy.ndarray): A dict of arbitrary agent
+                    state information.
+                * step_type (garage.StepType): a `StepType` enum value. Can
+                    either be StepType.FIRST, StepType.MID, StepType.LAST,
+                    or StepType.TIMEOUT.
+
+        """
+
+    @abc.abstractmethod
+    def render(self, mode):
+        """Renders the environment.
+
+        The set of supported modes varies per environment. (And some
+        environments do not support rendering at all.) By convention,
+        if mode is:
+
+        * display: render to the current display or terminal and
```

Comment:
I definitely think it doesn't make sense to have this be part of the same method as the other options (even though gym does it this way).